src/index.ts

```diff
@@ -592,12 +592,20 @@ class Logging {
     }
     delete reqOpts.autoPaginate;
     delete reqOpts.gaxOptions;
-    const gaxOptions = extend(
+    let gaxOptions = extend(
       {
         autoPaginate: options!.autoPaginate,
       },
       options!.gaxOptions
     );
+    if (options?.maxResults) {
+      gaxOptions = extend(
+        {
+          maxResults: options.maxResults,
+        },
+        gaxOptions
+      );
+    }
```

Comment:
nit: why not to update only the `maxResults` field? Unless it is a case of the dynamic structure when the code operates with typeless architecture?

Reply:
Indeed - we operate here on typeless structure